### DISCUSSION TITLE
Enable upper-case characters in email alias

### DISCUSSION THREAD

**Discussion by WhamboMPS** [simple-login/app#563] [Feature Requests]
For instance, `SimpleLogin@aleeas.com` instead of `simplelogin@aleeas.com`.

Yes, I know that email addresses are inherently case insensitive in the RFC. However, it is humans that need to read them. And being able to have mixed casing improves readability.

Implementation idea: I know that mixed casing can cause problems with sorting in some instances. The way that engineers that I worked with in a previous life handled this situation was to store the data in two fields. The first was _as entered_ and the second was a copy of that data _forced to lower-case._ The as entered version was used for the UI and the lower case version was used for sorting and was never shown to the user.

**Comment by nguyenkims** [comment]
This is indeed a good idea. Thanks for sharing the tips about having 2 fields for storing email addresses, this is indeed the best way to handle uppercase characters in email addresses.

**Reply by WhamboMPS** [reply]
@nguyenkims: This is what those of us who don't know how to program do: we retell the good ideas of the brilliant programmers we were fortunate enough to work alongside. :-)

**Comment by WhamboMPS** [comment]
@nguyenkims: any thoughts on whether this feature will be implemented?

**Reply by nguyenkims** [reply]
@WhamboMPS hey our todo list is quite packed at the moment and unfortunately this ticket isn't part of our next items ... As the alias is now quite short, having uppercase letter in the aliases is maybe not necessary anymore, what do you think?

**Comment by mr-thomas-z** [comment]
My only suggestion is to make sure that upper case I (as in Italy), lower case l (as in lucy), and 1 can be easily distinguished, or else it'd be inconvenient to spell out the email address over the phone.

The same goes for O (as in Olivia) and 0.

**Reply by nguyenkims** [reply]
Do you think using a monospaced font for email aliases would solve this problem?
